cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -0,0 +1,137 @@
+		Use:   "plan",
+		Short: "Check which versions are available to upgrade to and validate whether your current cluster is upgradeable",
+		Run: func(_ *cobra.Command, _ []string) {
+			err := RunPlan(parentFlags)
+			kubeadmutil.CheckErr(err)
+		},
+	}
+
+	return cmd
+}
+
+// RunPlan takes care of outputting available versions to upgrade to for the user
+func RunPlan(parentFlags *cmdUpgradeFlags) error {
+
+	// Start with the basics, verify that the cluster is healthy, build a client and a versionGetter. Always do this non-interactively for plan
+	upgradeVars, err := EnforceRequirements(parentFlags.kubeConfigPath, parentFlags.cfgPath, parentFlags.printConfig, true)
+	if err != nil {
+		return err
+	}
+
+	// Compute which upgrade possibilities there are
+	availUpgrades, err := upgrade.GetAvailableUpgrades(upgradeVars.versionGetter, parentFlags.allowExperimentalUpgrades, parentFlags.allowRCUpgrades)
+	if err != nil {
+		return err
+	}
+
+	// Tell the user which upgrades are available
+	printAvailableUpgrades(availUpgrades, os.Stdout)
+	return nil
+}
+
+// printAvailableUpgrades prints a UX-friendly overview of what versions are available to upgrade to
+func printAvailableUpgrades(upgrades []upgrade.Upgrade, w io.Writer) {
+
+	// Return quickly if no upgrades can be made
+	if len(upgrades) == 0 {
+		fmt.Fprintln(w, "Awesome, you're up-to-date! Enjoy!")
+		return
+	}
+	// The tab writer writes to the "real" writer w
+	tabw := tabwriter.NewWriter(w, 10, 4, 3, ' ', 0)
+
+	// Loop through the upgrade possibilities and output text to the command line
+	for _, upgrade := range upgrades {
+
+		if upgrade.CanUpgradeKubelets() {
+			fmt.Fprintln(w, "Components that must be upgraded manually after you've upgraded the control plane with 'kubeadm upgrade apply':")
+			fmt.Fprintln(tabw, "COMPONENT\tCURRENT\tAVAILABLE")
+			firstPrinted := false
+
+			// The map is of the form <old-version>:<node-count>. Here all the keys are put into a slice and sorted
+			// in order to always get the right order. Then the map value is extracted separately
+			for _, oldVersion := range sortedSliceFromStringIntMap(upgrade.Before.KubeletVersions) {
+				nodeCount := upgrade.Before.KubeletVersions[oldVersion]
+				if !firstPrinted {
+					// Output the Kubelet header only on the first version pair
+					fmt.Fprintf(tabw, "Kubelet\t%d x %s\t%s\n", nodeCount, oldVersion, upgrade.After.KubeVersion)
+					firstPrinted = true
+					continue
+				}
+				fmt.Fprintf(tabw, "\t\t%d x %s\t%s\n", nodeCount, oldVersion, upgrade.After.KubeVersion)
+			}
+			// We should flush the writer here at this stage; as the columns will now be of the right size, adjusted to the above content
+			tabw.Flush()
+			fmt.Fprintln(w, "")
+		}
+
+		fmt.Fprintf(w, "Upgrade to the latest %s:\n", upgrade.Description)
+		fmt.Fprintln(w, "")
+		fmt.Fprintln(tabw, "COMPONENT\tCURRENT\tAVAILABLE")
+		fmt.Fprintf(tabw, "API Server\t%s\t%s\n", upgrade.Before.KubeVersion, upgrade.After.KubeVersion)
```

Comment:
Might want to consider https://github.com/ryanuber/columnize here, or do similar wrapping.